dio/lib/src/dio_event_processor.dart

```diff
@@ -0,0 +1,110 @@
+// ignore_for_file: implementation_imports, public_member_api_docs
+
+import 'dart:async';
+
+import 'package:dio/dio.dart';
+import 'package:sentry/sentry.dart';
+import 'package:sentry/src/sentry_exception_factory.dart';
+import 'package:sentry/src/sentry_stack_trace_factory.dart';
+
+class DioEventProcessor extends EventProcessor {
+  DioEventProcessor(this._options, this._maxRequestBodySize);
+
+  final SentryOptions _options;
+  final MaxRequestBodySize _maxRequestBodySize;
+  late final sentryExceptionFactory = SentryExceptionFactory(
+    _options,
+    SentryStackTraceFactory(_options),
+  );
+
+  @override
+  FutureOr<SentryEvent?> apply(SentryEvent event, {dynamic hint}) {
+    final dynamic dioError = event.throwable;
+    if (dioError is! DioError) {
+      return event;
+    }
+
+    try {
+      final mechanism = Mechanism(
+        type: 'DioIntegration',
+        description: 'This is the inner exception of the DioError',
```

Comment:
But it should still show the original Mechanism for the DioError. 

@marandenato How do order the exceptions, so that the DioError is the main exception and the the inner exception is the second?

Reply:
It always shows `runZonedGuarded` for the `DioError`.
But not for the cause when the description is set, see https://github.com/getsentry/sentry-dart/pull/718#issuecomment-1023036545

Reply:
Might be something buggy in the UI.

Reply:
the sorting matter, See `SentryStackTraceFactory#getStackFrames`, we `reversed` the list because of that.
`DioError` should be the last item on the exception list.